core/src/software/aws/toolkits/core/utils/Waiter.kt

```diff
@@ -3,14 +3,87 @@
 
 package software.aws.toolkits.core.utils
 
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.runBlocking
 import software.amazon.awssdk.awscore.exception.AwsServiceException
 import java.time.Duration
+import java.time.Instant
+import java.util.concurrent.CompletionException
+import kotlin.math.min
+import kotlin.reflect.KClass
+
+object Waiters {
+    private val LOG = getLogger<Waiters>()
+
+    /**
+     * Generic waiter method to wait for a certain AWS resource to be in a steady status until it reaches in a failed status
+     * or time out for pulling.
+     * @param T The AWS resource type for querying the status from it
+     */
+    fun <T> waitUntil(
```

Comment:
Do we want to make our waiter suspend then allow callers to runBlocking if they want?

Reply:
Not sure, puts more bloat into the call sites.... seems like the name would indicate it should always block

Reply:
it puts more bloat on non-coroutine call sites, but shouldn't this be called from coroutines like 90%+ of the time?